### DISCUSSION TITLE
Automatic favicon generation support

### DISCUSSION THREAD

**Discussion by gergelypap** [vercel/next.js#21926] [Ideas]
**Describe the feature you'd like to request**

NextJS should support automatic favicon and touch-icon generation, based on a configuration file, similarly to what https://realfavicongenerator.net/ can achieve.

The idea is to provide a single image (SVG, JPG, PNG, whatever), and everything would be automagically generated from that. Configuration should allow setting android touch icons, safari pinned tab icon, simple favicon.ico, windows tile icons via browserconfig.xml, and maybe even a site.webmanifest file.

**Describe the solution you'd like**

I would imagine the solution as a `next.config.js` configuration key, such as:

```javascript
module.exports = {
  appIcon: {
    file: 'path/to/icon.svg',
    themeColor: '#000000',
    msTileColor: '#000000',
    safariPinnedTabColor: '#000000',
    sizes: [192, 512],
    // etc...
  }
}
```

After the build, the required meta tags would be added to `<head>`, and icon files would be generated in a pre-configured place. Probably these files should be generated in the `public` directory.

```
android-chrome-192x192.png
android-chrome-512x512.png
apple-touch-icon.png
browserconfig.xml
favicon-16x16.png
favicon-32x32.png
favicon.ico
mstile-144x144.png
mstile-150x150.png
mstile-310x150.png
mstile-310x310.png
mstile-70x70.png
safari-pinned-tab.svg
site.webmanifest
```

Now, `site.webmanifest` is a special case, because it is required for a PWA as well. Maybe it could be parsed and updated accordingly, if present.

Yes, I know that https://www.npmjs.com/package/favicons is exactly doing this, but maybe NextJS could integrate it and make its configuration simpler, while injecting the meta tags as well.

**Comment by agusterodin** [comment]
I wish they had it out of the box as well (or some sort of officially sanctioned solution), something similar to gatsby-plugin-manifest. In the meantime here are two solutions:

https://github.com/agusterodin/Modern-Favicons (made by me, very lightweight)
https://github.com/tinialabs/next-favicon-loader (more of a gatsby-plugin-manifest equivalent)
